lib/saluki-components/src/sources/dogstatsd/framer.rs

```diff
@@ -1,26 +1,31 @@
 use saluki_io::{
-    deser::{
-        codec::DogstatsdCodec,
-        framing::{LengthDelimitedFramer, NewlineFramer},
-    },
-    multi_framing,
+    deser::framing::{Framer, LengthDelimitedFramer, NewlineFramer},
     net::ListenAddress,
 };
 
-use super::interceptor::AgentLikeTagMetadataInterceptor;
+pub enum DsdFramer {
+    Newline(NewlineFramer),
+    LengthDelimited(LengthDelimitedFramer),
+}
 
-multi_framing!(name => DogStatsD, codec => DogstatsdCodec<AgentLikeTagMetadataInterceptor>, {
-    Newline => NewlineFramer,
-    LengthDelimited => LengthDelimitedFramer,
-});
+impl Framer for DsdFramer {
+    fn next_frame<'a, B: saluki_io::buf::ReadIoBuffer>(
```

Comment:
Small nit: we should use imports rather than fully-qualified paths.

(Gotta make a mental note to see if I can make `rustfmt` check/fix this for us automatically)

Reply:
Agreed, that was an editor oopsie.